### DISCUSSION TITLE
Set Twig template for account activation email or password reset

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#8855] [Ideas]
Right now we can only replace the static translation string.

**Comment by brandonkelly** [comment]
Can you provide more details? Not sure I’m following you.

**Comment by piotrpog** [comment]
@brandonkelly 
right now these emails are sent as plaintext and their content is set in static file translations.
I would like to use the twig template as their content, for example to have them use such templates:
https://postmarkapp.com/transactional-email-templates

**Comment by brandonkelly** [comment]
So to be clear you are asking that all system emails have the ability to assign their own Twig templates that could be used for the HTML emails?

**Comment by piotrpog** [comment]
@brandonkelly that sums it up pretty nicely.

**Comment by piotrpog** [comment]
@brandonkelly 
Today I tested system messages functionality a bit. Although we can use Twig code in these mesages, the whole message is wrapped in basic HTML tags upon sending. This means that I cannot use `{% extends %}` to wrap message with some email theme, because html tags will be duplicated.

**Comment by ccchapman** [comment]
In my case, I decided to set the `HTML Email Template` setting to a Twig template `_email.twig` and perform any logic in there. This gives you [access](https://docs.craftcms.com/api/v3/craft-models-systemmessage.html#public-properties) to the message's key, subject, body, etc.

```twig
{% set message = craft.app.systemMessages.getMessage(emailKey) %}
```

`emailKey` is passed along with any system messages Craft sends out. `link` and `user` variables are also included for many of the messages.

You may also have it render Twig declared in the SystemMessage's body.

```twig
{{ include(template_from_string(message.body))|nl2br }}
```
